auth/credentials/impersonate/idtoken.go

```diff
@@ -86,20 +93,21 @@ func NewIDTokenCredentials(opts *IDTokenOptions) (*auth.Credentials, error) {
 
 	client := opts.Client
 	creds := opts.Credentials
+	logger := internallog.New(opts.Logger)
 	if client == nil {
 		var err error
-		if creds == nil {
-			// TODO: test not signed jwt more
-			creds, err = credentials.DetectDefault(&credentials.DetectOptions{
-				Scopes:           []string{defaultScope},
-				UseSelfSignedJWT: true,
-			})
-			if err != nil {
-				return nil, err
-			}
+		// TODO: test not signed jwt more
+		creds, err = credentials.DetectDefault(&credentials.DetectOptions{
```

Comment:
Will this overwrite the value from `opts.Credentials` (if present) set above on line 95?

Reply:
Maybe `opts.Credentials` could be passed in here using `DetectOptions`, and line 95 deleted?

Reply:
This was just a bad merge on my part when I rebased. Fixed it up.